### PR TITLE
Task/rhornung67/kernel docs

### PR DESCRIPTION
**Pull request by rhornung67**
Expanded and improved discussion of kernel statement types (feature section) and kernel usage (tutorial section and code examples for matrix multiplication operation).

Adam, adding you as a reviewer on this since I would like you to scrutinize my documentation of kernel features. This is the first chunk -- there is more to come.

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
This looks good!

File: docs/sphinx/user_guide/feature/loop_basic.rst [hunk 1]
```diff
@@ -68,35 +62,64 @@ may be written in a RAJA form as::
...
   });
 
-``RAJA::forall`` templates take a template argument for the 
-execution policy and two arguments: an object describing the loop iteration 
-space, such as a RAJA segment or index set, and a lambda expression defining 
-the loop body. Applying different loop execution policies enables the loop to 
-run in different ways; e.g., using different programming model back-ends. 
-Different iteration space objects, enable the loop iterates to be reordered, 
-run in different threads, etc. 
+A ``RAJA::forall`` template is templated on an execution policy type and takes
```
Comment (artv3) [inline docs/sphinx/user_guide/feature/loop_basic.rst]: ``template is templated" sounds a bit repetitive could we say "A ``RAJA::forall`` method is templated ..."?

Comment (rhornung67) [reply docs/sphinx/user_guide/feature/loop_basic.rst]: Thanks. I'll change that.

**Comment by artv3** [comment]
I only had one minor comment, but the more I spend time with kernel the more I like it! I think these notes capture the building blocks of kernel well.

**Comment by rhornung67** [comment]
@artv3 I just pushed a wording change based on your comment. Let me know when you're good with this and I'll merge it.  Thanks.

**Pull request merged**